src/compiler/program.ts

```diff
@@ -3160,7 +3160,7 @@ namespace ts {
                 createDiagnosticForOptionName(Diagnostics.Option_0_cannot_be_specified_when_option_target_is_ES3, "useDefineForClassFields");
             }
 
-            if (options.checkJs && !options.allowJs) {
+            if (options.checkJs && !getAllowJSCompilerOption(options)) {

```

Comment:
This seems incorrect condition. You want to check if options.allowJs === false here instead

Reply:
Duh, thanks for the catch.

Reply:
No, wait, that undoes the loosening from this PR. THe current code only errors when `checkJs: true` but `allowJs: false` explicitly.

Reply:
```suggestion
            if (options.checkJs && options.allowJs === false) {
```
will also work and won’t require a function call.